File: app/src/main/java/fr/free/nrw/commons/media/Caption.java
package fr.free.nrw.commons.media;

import com.google.gson.annotations.SerializedName;

/**
 * Model class for parsing Captions when fetching captions using filename in MediaClient
 */
public class Caption {

    /**
     * users language in which caption is written
     */
    @SerializedName("language")
    private String language;
    @SerializedName("value")
    private String value;

    /**
     * No args constructor for use in serialization
     */
    public Caption() {
    }

    /**
     * @param value
     * @param language
     */
    public Caption(String language, String value) {
        super();
        this.language = language;
        this.value = value;
    }

    @SerializedName("language")
    public String getLanguage() {
        return language;
    }

    @SerializedName("value")
    public String  getValue() {
        return value;
    }
}

File: app/src/main/java/fr/free/nrw/commons/media/Caption.kt
package fr.free.nrw.commons.media

import com.google.gson.annotations.SerializedName

/**
 * Model class for parsing Captions when fetching captions using filename in MediaClient
 */
class Caption() {
    @SerializedName("language")
    var language: String? = null

    @SerializedName("value")
    var value: String? = null

    constructor(language: String?, value: String?) : this() {
        this.language = language
        this.value = value
    }
}
